plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/actions/CollectDiagnosticInfoHandler.java

```diff
@@ -195,7 +196,11 @@ protected Composite createDialogArea(Composite parent) {
             );
             textWithOpen.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
             String userHome = System.getProperty(StandardConstants.ENV_USER_HOME);
-            if (userHome != null) {
+            if(folderPath.length()>0 && !folderPath.equals(userHome))
+            {
+                textWithOpen.setText(folderPath);
+            }
+            else if (userHome != null) {
```

Comment:
Instead of having this cumbersome logic, how about using it as the currently selected path? Assign `System.getProperty(StandardConstants.ENV_USER_HOME)` to it and change it inside the `okPressed` method.